### DISCUSSION TITLE
Godot engine for future versions?

### DISCUSSION THREAD

**Discussion by DanielLester83** [maforget/ComicRackCE#25] [Ideas]
Not sure how far you'd want to change Comicrack for porting or new features, but I saw that the Godit engine is expanding support for C# and .NET see here: https://godotengine.org/article/platform-state-in-csharp-for-godot-4-2/

**Comment by maforget** [comment]
This is cool as an alternative to Unity, but I fail to catch how a game engine expanding support to C# will have anything to do with ComicRack. If we would need to redo the UI, I wouldn't use a game engine. Probably something like Avalonia or the new MAUI from Microsoft.

**Reply by DanielLester83** [reply]
I suppose it would come down to what people are more comfortable with or enthusiastic about.

**Reply by Totengeist** [reply]
I think a game engine would be overkill. It might include too many things we're not going to use, which might increase file size or complexity. I think we should use what's right for the job, unless no one can/will. A framework like Avalonia or Eto Forms makes more sense to me, but I'm not experienced in UI programming outside of Python or PHP.

**Comment by apiweb** [comment]
I believe Godot doesn't align well with the interface. 
ComicRack, being a desktop application utilizing Windows Forms, is inherently designed for creating Windows applications natively. Migrating to Godot or any game engine implies the need to recreate all components for interaction, such as buttons, lists, dropdowns, etc.

As maforget pointed out, we now have more suitable alternatives like Avalonia and Maui. 
I personally prefer Avalonia due to its broader multi-platform possibilities, extensive support, well-documented resources, and greater recognition within the community. 
On the other hand, I've encountered challenges with Maui; it seems more oriented towards mobile development rather than desktop, and I sense it may be abandoned by Microsoft in the near future.

However, delving into this discussion is quite advanced. Initially, it's necessary to port the existing codebase to .NET Core, for example.

**Comment by Totengeist** [comment]
> Initially, it's necessary to port the existing codebase to .NET Core, for example.

How extensive do you estimate that to be? I'm relatively new to C# and .NET, so it's difficult to get a good sense of it. At what point would you say it is worth investing the time and energy into this?

Based on discussions on Reddit and elsewhere, I think there's a demand for ComicRack on other platforms, but it's hard to gauge if that's a significant portion of the userbase.

**Reply by DanielLester83** [reply]
I'd personally like to see a version for the steamdeck (linux) eventually.
